### PR TITLE
feat: add a pipeline which scans the Go dependencies with whitesource

### PR DESCRIPTION
**Pull request by ccojocar**
#### Submitter checklist

- [x] Change is code complete and matches issue description.
- [ ] Change is covered by existing or new tests.

#### Description

This adds a new pipeline which will scan the Go dependencies using the whitesource agent. This will execute
when a pull request is merged to master and it will run in parallel with the default release pipeline.

The dependencies are updated to whitesource service which will check if there is any CVE open for them. 
Note that currently this will only report the dependencies, there isn't any policy in place to fail the build in 
case there is any critical/high CVE.

#### Special notes for the reviewer(s)

This requires https://github.com/jenkins-x/environment-tekton-weasel-dev/pull/405 in oder to run.

#### Which issue this PR fixes

fixes #

### CODE REVIEW

File: whitesource.config [hunk 1]
```diff
@@ -0,0 +1,256 @@
+####################################################################
```
Comment (hferentschik) [inline whitesource.config]: what do you think about making this config a dot file or even better imo move the config into _hacking_. I really would like to reduce the clutter in the root directory.

Comment (ccojocar) [reply whitesource.config]: This is the default config provided by whitesource which is loaded from the root folder by the container which runs in the pipeline.

Comment (hferentschik) [reply whitesource.config]: and the location is not configurable?

Comment (hferentschik) [reply whitesource.config]: Will there be a change to the Prow config as well? Do you want this as a required status check? Will this even run on our prod cluster?

Comment (ccojocar) [reply whitesource.config]: The PR which changes the dev environment is linked in the description. 

https://github.com/jenkins-x/environment-tekton-weasel-dev/pull/405

Comment (ccojocar) [reply whitesource.config]: I guess that I can make the config path configurable  and when is not set can default to `.whitesource.config`

Comment (hferentschik) [reply whitesource.config]: > I guess that I can make the config path configurable and when is not set can default to .whitesource.config

that would be awesome

Comment (ccojocar) [reply whitesource.config]: I  changed the whitesource image to make to be able to setup the configuration path.

Comment (ccojocar) [reply whitesource.config]: Please could you also review the prow pull request? I think this can be merged first.

**Pull request merged**